frontend/src/models/admin.js

```diff
@@ -216,6 +216,26 @@ const Admin = {
         return { success: false, error: e.message };
       });
   },
+  setWelcomeMessages: async function (messages) {
+    return fetch(`${API_BASE}/system/set-welcome-messages`, {
+      method: "POST",
+      headers: {
+        ...baseHeaders(),
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ messages }),
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(res.statusText || "Error setting welcome messages.");
+        }
+        return { success: true, ...res.json() };
```

Comment:
if success is a tag that should exist, return it from the response then. Can then just simplify this to
```
.then((res) => {
  if(!res.ok) throw....
  return res.json()
}
```